File: 1765-merge-in-between-linked-lists/merge-in-between-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {number} a
 * @param {number} b
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeInBetween = function(list1, a, b, list2) {
    // Find the node before the removal range
    let prev = null;
    let curr = list1;
    for (let i = 0; i < a; i++) {
        prev = curr;
        curr = curr.next;
    }

    // Remove the nodes in the range [a, b]
    let nextNode = curr.next;
    for (let i = 0; i < b - a; i++) {
        curr.next = nextNode;
        curr = nextNode;
        if (nextNode) {
            nextNode = nextNode.next;
        }
    }

    // Insert list2 into the removed range
    if (prev) {
        prev.next = list2;
    } else {
        list1 = list2;
    }
    let curr2 = list2;
    while (curr2.next) {
        curr2 = curr2.next;
    }
    curr2.next = nextNode;

    return list1;
};

File: 1765-merge-in-between-linked-lists/merge-in-between-linked-lists.py
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        # Find the node before the removal range
        prev = None
        curr = list1
        for _ in range(a):
            prev = curr
            curr = curr.next
        
        # Remove the nodes in the range [a, b]
        next_node = curr.next
        for _ in range(b - a):
            curr.next = next_node
            curr = next_node
            if next_node:
                next_node = next_node.next
        
        # Insert list2 into the removed range
        if prev:
            prev.next = list2
        else:
            list1 = list2
        curr = list2
        while curr.next:
            curr = curr.next
        curr.next = next_node
        
        return list1